gateway-server/src/main/java/org/apache/knox/gateway/services/token/impl/DefaultTokenAuthorityService.java

```diff
@@ -86,6 +89,8 @@
   private byte[] cachedSigningHmacSecret;
   private RSAPrivateKey signingKey;
 
+  private Optional<String> cachedSigningKeyKID = Optional.empty();
```

Comment:
`cachedSigningKeyID` ?

Reply:
ah, the silly KeyKid will change it, thanks!